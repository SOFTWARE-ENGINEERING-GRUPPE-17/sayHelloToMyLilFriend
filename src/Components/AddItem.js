import React from 'react';
import PropTypes from 'prop-types';
import Dropzone from "react-dropzone-uploader";


export const AddItem = ({name, price, image, onChange, onSubmit}) => (

    <div className="row justify-content-center">
        <form className="form-inline" onSubmit={onSubmit}>
            <input
                type="text"
                className="form-control mb-2 mr-sm-2"
                placeholder="Adresse"
                value={name}
                name="name"
                onChange={onChange}
            />

            <div className="input-group mb-2 mr-sm-2">
                <input
                    type="text"
                    className="form-control"
                    placeholder="Pris"
                    value={price}
                    name="price"
                    onChange={onChange}
                />
                <br/>
                <input
                    type='file'
                    name={"image"}
                    value={image}
                    onChange={onChange}
                />
            </div>
            <button type="submit" className="btn btn-primary mb-2 pxy-4">Legg til</button>
        </form>
    </div>
);

AddItem.propTypes = {
    name: PropTypes.string.isRequired,
    price: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    onSubmit: PropTypes.func.isRequired,
};
