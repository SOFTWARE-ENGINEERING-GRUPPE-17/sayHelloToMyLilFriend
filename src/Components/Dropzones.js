import React, { useState } from "react";

import Dropzone from "react-dropzone";

export default function Dropzones() {
    const [fileNames, setFileNames] = useState([]);
    const handleDrop = acceptedFiles =>
        setFileNames(acceptedFiles.map(file => file.name));

    return (
        <div className="Dropzone_tester">
            <Dropzone
                onDrop={handleDrop}
                accept="image/*"
                minSize={1024}
                maxSize={3072000}
            >
                {({ getRootProps, getInputProps }) => (
                    <div {...getRootProps({ className: "dropzone" })}>
                        <input {...getInputProps()} />
                        <p>Drag'n'drop images, or click to select files</p>
                    </div>
                )}
            </Dropzone>
            <div>
                <strong>Files:</strong>
                <ul>
                    {fileNames.map(fileName => (
                        <li key={fileName}>{fileName}</li>
                    ))}
                </ul>
            </div>
        </div>
    );
}