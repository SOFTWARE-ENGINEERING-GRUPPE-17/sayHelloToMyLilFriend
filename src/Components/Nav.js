import React from 'react';
import {Link} from "react-router-dom";
import '../App.css';

const navStyle = {
    color: "white"
};

export const Nav = () => (
    <nav>
        <h3>Parkering AS</h3>
        <ul className="nav-links">
            <Link style={navStyle}  to="/Home">
                <li>Hjem</li>
            </Link>
            <Link style={navStyle} to="/Park">
                <li>Ledige parkeringer</li>
            </Link>
            <Link style={navStyle}  to="/Rent">
                <li>Lei ut parkering</li>
            </Link>

        </ul>
    </nav>
);
export default Nav;