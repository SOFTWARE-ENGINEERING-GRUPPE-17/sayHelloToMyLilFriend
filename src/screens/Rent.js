
import React, {Component} from 'react';
//import image from '../images/1.png';
import {ItemCard} from "../Components/ItemCard";
import {AddItem} from "../Components/AddItem";



class Rent extends Component {
    state = {
        image: "",
        name: "",
        price: "",
        items: []
    };

    handleInputChange = event => {
        const target = event.target;
        const value = target.value;
        const name = target.name;
        this.setState({
            [name]: value
        });
    };

    addItem = event => {
        event.preventDefault();
        const {name, price, image} = this.state;
        const itemsInState = this.state.items;
        const itemsArrayLength = itemsInState.length;
        const id = itemsArrayLength ? (itemsInState[itemsArrayLength - 1].id + 1) : 1;
        this.setState({
            items: [
                ...itemsInState,
                Object.assign({}, {
                    id,
                    name,
                    price,
                    image
                })
            ],
            name: "",
            price: "",
            image: ""
        })
    };

    toggleItemEditing = index => {
        this.setState({
            items: this.state.items.map((item, itemIndex) => {
                if (itemIndex === index) {
                    return {
                        ...item,
                        isEditing: !item.isEditing
                    }
                }
                return item;
            })
        });
    };

    handleItemUpdate = (event, index) => {
        const target = event.target;
        const value = target.value;
        const name = target.name;
        this.setState({
            items: this.state.items.map((item, itemIndex) => {
                if (itemIndex === index) {
                    return {
                        ...item,
                        [name]: value
                    }
                }
                return item;
            })
        });
    };


    onDelete = index => {
        this.setState({
            items: [
                ...this.state.items.slice(0, index),
                ...this.state.items.slice(index + 1)
            ]
        });
    };




    render() {
        const {name, price, image   } = this.state;
        return <div>

            <div className="container pt-4">

                <AddItem
                    name={name}
                    price={price}
                    image={image}
                    onChange={this.handleInputChange}
                    onSubmit={this.addItem}
                />

                <h1 className="display-4 my-4 text-center text-muted">Items</h1>

                <div className="row">
                    {
                        this.state.items.map((item, index) =>
                            <ItemCard
                                key={item.id}
                                index={index}
                                image={image}
                                item={item}
                                toggleEditing={() => this.toggleItemEditing(index)}
                                onChange={this.handleItemUpdate}
                                onDelete={() => this.onDelete(index)}
                            />
                        )
                    }
                </div>

                <hr/>
            </div>
        </div>;
    }
}

export default Rent;