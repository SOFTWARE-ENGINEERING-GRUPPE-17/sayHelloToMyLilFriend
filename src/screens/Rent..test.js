import React from 'react'
import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import Rent from "./Rent";
import Home from "./HomeScreen";
import Park from "./Park";
import {ItemCard} from "../Components/ItemCard";
import {AddItem} from "../Components/AddItem";

describe('Rent Component', () => {

  beforeAll(() => {
    Enzyme.configure({ adapter: new Adapter() })
  })

  it('renders correctly', () => {
    const component = shallow(<Rent/>)
    expect(component).toMatchSnapshot()
  })

  it("Register form exists", () => {
    const wrapper = shallow(<Park/>);
    const formHeader = <h1 className="display-4 my-4 text-center text-muted">Parkeringer</h1>;
    expect(wrapper.contains(formHeader)).toEqual(true);
    expect(wrapper.contains(AddItem)).toEqual(true);
    expect(wrapper.contains(ItemCard)).toEqual(true);
    // expect(wrapper.find("Card.Card1").text());

  });

  it('Registered form gets saved', () => {
    const creator = mount(<AddItem name="adresse" label="name" price="price"/>);
    const label = creator.find('label')
    expect(label).toHaveLenth(1);
    expect(label.prop('htmlFor')).toEqual('adresse');
    expect(label.text()).toEqual('adresse');

    const input = creator.find('input');
    expect(input).toHaveLenth(1);
    expect(input.prop('type')).toEqual('text');
    expect(input.prop('name')).toEqual('adresse');
    expect(input.prop('price')).toEqual('price');

  });

})
