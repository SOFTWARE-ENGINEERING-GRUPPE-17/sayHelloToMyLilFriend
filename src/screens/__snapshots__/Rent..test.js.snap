// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rent Component Register form exists 1`] = `ShallowWrapper {}`;

exports[`Rent Component renders correctly 1`] = `ShallowWrapper {}`;
